versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapterConfig.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.mongodb;
+
+import org.immutables.value.Value;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig;
+
+@Value.Immutable(lazyhash = true)
+public interface MongoDatabaseAdapterConfig extends DatabaseAdapterConfig {
+
+  String getConnectionString();
+
+  String getDatabaseName();
+
+  @Value.Derived
```

Comment:
I suspect this should be `@Value.Default` ("derived" means it's always built, "default" means its built when not explicitly set)

Reply:
That's what I meant actually. `Derived` -> built on first get and cached. Since the constructor is trivial, I thought it'd make sense instead of forcing it to be created and set from the outside. WDYT?

Reply:
Immutables takes care of that piece in the generated code.

Reply:
Yes, so "derived" is created only on first access. Why is "default" preferable?

Reply:
Made `Client` settable in d7a1e320814af38cdfd0b5d4c75487e841fd7de9